src/main/java/com/marklogic/client/ext/datamovement/consumer/WriteRowToTableauConsumer.java

```diff
@@ -0,0 +1,250 @@
+package com.marklogic.client.ext.datamovement.consumer;
+
+import com.marklogic.client.MarkLogicIOException;
+import com.marklogic.client.example.cookbook.datamovement.TypedRow;
+import com.marklogic.client.type.*;
+import com.tableausoftware.TableauException;
+import com.tableausoftware.common.Collation;
+import com.tableausoftware.extract.Extract;
+import com.tableausoftware.extract.Row;
+import com.tableausoftware.extract.TableDefinition;
+import com.tableausoftware.extract.Table;
+import com.tableausoftware.common.Type;
+
+import java.io.File;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+
+
+import java.util.function.Consumer;
+
+/**
+ * This is a Consumer which takes in a TypedRow and writes the row data
+ * extracted via ExtractViaTemplateListener to a file of format Tableau Data
+ * Extract. It can be initialized as follows:
+ *
+ * <pre>{@code
+ * WriteRowToTableauConsumer tableauWriter = new WriteRowToTableauConsumer("output.tde")
+ *    .withColumn("firstName", Type.UNICODE_STRING)
+ *    .withColumn("salary", Type.INTEGER);
+ * }</pre>
+ *
+ * where output.tde file is the output Tableau Data Extract format file
+ * which contains the extracted rows. We should have the necessary columns
+ * defined with withColumn() method. The column names defined in the Template
+ * Driven Extraction templates should match the ones specified in the
+ * withColumn() method. Then it can be passed directly to the
+ * onTypedRowReady method of ExtractViaTemplateListener method as follows:
+ *
+ * <pre>{@code
+ * ExtractViaTemplateListener listner = new ExtractViaTemplateListener()
+ *      .withTemplate(templateUri)
+ *      .onTypedRowReady(tableauWriter)
+ * }</pre>
+ *
+ * This consumer depends on certain Tableau SDK for creating the Tableau Data
+ * Extract file. For more details on how to install the SDK and using the
+ * Tableau jars, please visit the following link:
+ *
+ * <a href="https://onlinehelp.tableau.com/current/api/sdk/en-us/help.htm">Tableau SDK</a>
+ * 
+ * This will work with MarkLogic server version greater than or equal to 9.0.5.
+ *
+ */
+public class WriteRowToTableauConsumer
+  implements Consumer<TypedRow>, AutoCloseable
+{
+  private boolean initialized = false;
+  private TableDefinition tableDef;
+  private LinkedHashMap<String,Column> columns = new LinkedHashMap<>();
+  private Extract extract;
+  private Table table;
+
+  // the tableauTdeFileName specifies the tableau data extract file we'll write to
```

Comment:
Use javadoc-style comments here so that they show up in javadocs (applies to each method that's currently using double slash comments)